src/main/java/talkie/components/Ui.java

```diff
@@ -0,0 +1,205 @@
+package talkie.components;
+
+import java.util.Scanner;
+
+import talkie.exception.TalkieException;
+import talkie.task.Task;
+import talkie.task.TaskList;
+
+/**
+ * Handles user interactions and displays messages to the user.
+ * <p>
+ * The {@code Ui} class manages the display of various types of messages to the user,
+ * such as welcome messages, task addition, deletion, and error handling.
+ * </p>
+ */
+public class Ui {
+
+    /**
+     * Enum representing different types of messages used by the {@code Ui} class.
+     */
+    public enum MessageType {
+        WELCOME_MESSAGE("Hello! I'm Talkie, your friendly ChatBot.\n"
+                + "What can I do for you?\n"),
+
+        BYE_MESSAGE("Bye. Hope to see you again soon!\n");
+
+        private final String message;
+
+        MessageType(String message) {
+            this.message = message;
+        }
+
+        /**
+         * Gets the message associated with the {@code MessageType}.
+         *
+         * @return The message string.
+         */
+        public String getMessage() {
+            return this.message;
+        }
+    }
+
+    // Scanner for input
+    private Scanner input = new Scanner(System.in);
+
+    /**
+     * Reads the next line of input from the user.
+     *
+     * @return The user's input as a string.
+     */
+    public String readCommand() {
+        return this.input.nextLine();
+    }
+
+    /**
+     * Closes the {@code Scanner} used for user input.
+     */
+    public void closeInput() {
+        this.input.close();
+    }
+
+    /**
+     * Displays a welcome message to the user.
+     */
+    public String welcomeMessage() {
+        return MessageType.WELCOME_MESSAGE.getMessage();
+    }
+
+    /**
+     * Displays a goodbye message to the user.
+     */
+    public String byeMessage() {
+        return MessageType.BYE_MESSAGE.getMessage();
+    }
+
```

Comment:
The names for these two methods are more like nouns instead of verbs. Perhaps welcome() and bye() will be more like method names.